pkg/plugins/resources/dockerimage/main.go

```diff
@@ -7,30 +7,36 @@ import (
 	"github.com/mitchellh/mapstructure"
 	"github.com/updatecli/updatecli/pkg/plugins/utils/docker/dockerimage"
 	"github.com/updatecli/updatecli/pkg/plugins/utils/docker/dockerregistry"
+	"github.com/updatecli/updatecli/pkg/plugins/utils/version"
 )
 
 // Spec defines a specification for a "dockerimage" resource
 // parsed from an updatecli manifest file
 type Spec struct {
-	// Architecture specifies the container image architecture such as `amd64`
+	// [S][C][T]Architecture specifies the container image architecture such as `amd64`
 	Architecture string `yaml:",omitempty"`
-	// Image specifies the container image such as `updatecli/updatecli`
+	// [S][C][T]Image specifies the container image such as `updatecli/updatecli`
 	Image string `yaml:",omitempty"`
-	// Tag specifies the container image tag such as `latest`
+	// [C][T]Tag specifies the container image tag such as `latest`
 	Tag string `yaml:",omitempty"`
-	// Username specifies the container registry username to use for authentication. Not compatible with token
+	// [S][C][T]Username specifies the container registry username to use for authentication. Not compatible with token
 	Username string `yaml:",omitempty"`
-	// Password specifies the container registry password to use for authentication. Not compatible with token
+	// [S][C][T]Password specifies the container registry password to use for authentication. Not compatible with token
 	Password string `yaml:",omitempty"`
-	// Token specifies the container registry token to use for authentication. Not compatible with username/password
+	// [S][C][T]Token specifies the container registry token to use for authentication. Not compatible with username/password
 	Token string `yaml:",omitempty"`
+	// [S]VersionFilter provides parameters to specify version pattern and its type like regex, semver, or just latest.
+	VersionFilter version.Filter `yaml:",omitempty"`
 }
 
 // DockerImage defines a resource of type "dockerimage"
 type DockerImage struct {
 	spec     Spec
 	registry dockerregistry.Registry
 	image    dockerimage.Image
+	// versionFilter holds the "valid" version.filter, that might be different than the user-specified filter (Spec.VersionFilter)
```

Comment:
In which case(s) are these values different?

Reply:
Due to Updatecli manifest templating, we avoid modifying spec data as it leads to unexpected data. So if we need to modify the object for some reason then we copy and work on the copied one.

Reply:
As a rule of thumb:
- Spec is a data specified by the end user through YAML (and eventually something else). It can be refreshed at any time and templatized.
- "Config" (naming is not always the same) is the transient data (in memory) with default configuration (default values, "Spec" converted to a custom model, and contextual data for dependency injection such as clients objects, etc.)